spec/requests/admin/admin_holidays_controller_spec.rb

```diff
@@ -65,5 +65,102 @@ module Admin::DiscourseCalendar
+          before do
+            sign_in(admin)
+          end
+
+          it "disables a specific holiday in a specific region" do
+            post "/admin/discourse-calendar/holidays/disable.json",
+              params: {
+                disabled_holiday: dia_de_la_independencia
+              }
+
+            expect(DiscourseCalendar::DisabledHoliday.last.disabled).to eq(true)
+            expect(DiscourseCalendar::DisabledHoliday.last.holiday_name).to eq(dia_de_la_independencia[:holiday_name])
+            expect(DiscourseCalendar::DisabledHoliday.last.region_code).to eq(dia_de_la_independencia[:region_code])
+          end
+
+          context "when a holiday has been added to the calendar" do
+            let(:calendar_post) { create_post(raw: "[calendar]\n[/calendar]") }
+            let(:australia_new_years_day) { { holiday_name: "New Year's Day", date: "2022-01-01", region_code: "au" } }
+            let(:australia_day) { { holiday_name: "Australia Day", date: "2022-01-26", region_code: "au" } }
+
+            before do
+              CalendarEvent.create(
+                topic_id: calendar_post.topic_id,
+                description: australia_new_years_day[:holiday_name],
+                start_date: australia_new_years_day[:date],
+                region: australia_new_years_day[:region_code]
+              )
+
+              CalendarEvent.create(
+                topic_id: calendar_post.topic_id,
+                description: australia_day[:holiday_name],
+                start_date: australia_day[:date],
+                region: australia_day[:region_code]
+              )
+            end
+
+            it "removes disabled holidays from the calendar" do
+              post "/admin/discourse-calendar/holidays/disable.json",
+                params: {
+                  disabled_holiday: {
+                    holiday_name: australia_new_years_day[:holiday_name],
+                    region_code: australia_new_years_day[:region_code],
+                  }
+                }
+
+              expect(CalendarEvent.where(
+                description: australia_new_years_day[:holiday_name],
+                region: australia_new_years_day[:region_code]
+              ).count).to eq(0)
+
+              expect(CalendarEvent.where(
+                description: australia_day[:holiday_name],
+                region: australia_day[:region_code]
+              ).count).to eq(1)
+            end
+          end
+        end
+      end
+    end
+
+    describe "#enable" do
+      context "when the calendar plugin is enabled" do
+        let(:calendar_enabled) { true }
+
+        context "when an admin is signed in" do
+          before do
+            sign_in(admin)
+          end
+
+          context "when there is a disabled holiday" do
+            let(:hong_kong_labour_day) { { holiday_name: "Labour Day", region_code: "hk" } }
+
+            before do
+              DiscourseCalendar::DisabledHoliday.create(hong_kong_labour_day)
```

Comment:
As a rule of thumb, we prefer the `!` version when calling create, save, or update.

Reply:
Thanks for mentioning this! I updated this and the other ones I added to use the `!` version.